docs/connector-development/config-based/stream-slicers.md

```diff
@@ -0,0 +1,172 @@
+# Stream Slicers
+
+`StreamSlicer`s define how to partition a stream into a subset of records.
+
+It can be thought of as an iterator over the stream's data, where a `StreamSlice` is the retriever's unit of work.
+
+When a stream is read incrementally, a state message will be output by the connector after reading every slice, which allows for [checkpointing](https://docs.airbyte.com/understanding-airbyte/airbyte-protocol/#state--checkpointing).
+
+At the beginning of a `read` operation, the `StreamSlicer` will compute the slices to sync given the connection config and the stream's current state,
+As the `Retriever` reads data from the `Source`, the `StreamSlicer` keeps track of the `Stream`'s state, which will be emitted after reading each stream slice.
+
+More information of stream slicing can be found in the [stream-slices section](../cdk-python/stream-slices.md)
+
+## Implementations
+
+This section gives an overview of the stream slicers currently implemented.
+
+### Datetime
+
+The `DatetimeStreamSlicer` iterates over a datetime range by partitioning it into time windows.
+This is done by slicing the stream on the records' cursor value, defined by the Stream's `cursor_field`.
+
+Given a start time, an end time, and a step function, it will partition the interval [start, end] into small windows of the size described by the step.
+For instance,
+
+```yaml
+stream_slicer:
+  start_datetime: "2021-02-01T00:00:00.000000+0000",
+  end_datetime: "2021-03-01T00:00:00.000000+0000",
+  step: "1d"
+```
+
+will create one slice per day for the interval `2021-02-01` - `2021-03-01`.
+
+The `DatetimeStreamSlicer` also supports an optional lookback window, specifying how many days before the start_datetime to read data for.
```

Comment:
as per slack discussion as something we should fix, we should also include that in the case of incremental syncs, the lookback window is also applied for the cursor

Reply:
created an issue https://github.com/airbytehq/airbyte/issues/15628